framework/framework.go

```diff
@@ -73,7 +73,7 @@ func (f *Framework) AddFunc(obj interface{}) {
 	defer f.mutex.Unlock()
 
 	ctx := context.Background()
-	ctx = cancelercontext.NewContext(ctx, make(chan struct{}, 1))
+	ctx = cancelercontext.NewContext(ctx, make(chan struct{}))
```

Comment:
We agreed to not buffer the channel since it is quite useless. The canceler channel is only closed to signal cancelation. See also the comments in the context package.